cdc/api/v2/api.go

```diff
@@ -14,25 +14,61 @@
 package v2
 
 import (
+	"net/http"
+
 	"github.com/gin-gonic/gin"
 	"github.com/pingcap/tiflow/cdc/api/middleware"
 	"github.com/pingcap/tiflow/cdc/capture"
+	"github.com/pingcap/tiflow/cdc/model"
+	"github.com/pingcap/tiflow/cdc/owner"
+	"github.com/pingcap/tiflow/pkg/upstream"
 )
 
 // OpenAPIV2 provides CDC v2 APIs
-type OpenAPIV2 struct {
+type openAPIV2 struct {
 	capture *capture.Capture
+	// use for unit test only
+	testStatusProvider owner.StatusProvider
+}
+
+// NewOpenAPIV2 creates a new openAPIV2.
+func NewOpenAPIV2(c *capture.Capture) openAPIV2 {
+	return openAPIV2{capture: c}
+}
+
+// NewOpenAPIV2ForTest returns a openAPIV2 for test
+func NewOpenAPIV2ForTest(c *capture.Capture, p owner.StatusProvider) openAPIV2 {
+	return openAPIV2{capture: c, testStatusProvider: p}
 }
 
-// NewOpenAPI creates a new openAPIs.
-func NewOpenAPI(c *capture.Capture) *OpenAPIV2 {
-	return &OpenAPIV2{capture: c}
+func (h *openAPIV2) statusProvider() owner.StatusProvider {
+	if h.testStatusProvider != nil {
+		return h.testStatusProvider
+	}
+	return h.capture.StatusProvider()
 }
 
-// RegisterOpenAPIRoutes registers routes for OpenAPI
-func RegisterOpenAPIRoutes(router *gin.Engine, api *OpenAPIV2) {
+// RegisterOpenAPIV2Routes registers routes for OpenAPI
+func RegisterOpenAPIV2Routes(router *gin.Engine, api openAPIV2) {
 	v2 := router.Group("/api/v2")
 
 	v2.Use(middleware.LogMiddleware())
 	v2.Use(middleware.ErrorHandleMiddleware())
+
+	// common APIs
+	v2.GET("/tso", api.GetTso)
+}
+
+func (h *openAPIV2) GetTso(c *gin.Context) {
+	ctx := c.Request.Context()
+	pdClient := h.capture.UpstreamManager.Get(upstream.DefaultUpstreamID).PDClient
```

Comment:
Can you check whether `UpstreamManager` or `UpstreamManager.Get(upstream.DefaultUpstreamID)` or `PDClient` is `nil`?

Reply:
ok